kubeadm/scripts/PrepareNode.ps1

```diff
@@ -0,0 +1,83 @@
+<#
+.SYNOPSIS
+Assists with preparing a Windows VM prior to calling kubeadm join
+
+.DESCRIPTION
+This script assists with joining a Windows node to a cluster.
+- Downloads Kubernetes binaries (kubelet, kubeadm) at the version specified
+- Registers wins as a service in order to run kube-proxy and cni as DaemonSets. More info here: 
+- Registers kubelet as an nssm service. More info on nssm: https://nssm.cc/
+
+.PARAMETER KubernetesVersion
+Kubernetes version to download and use
+
+.EXAMPLE
+PS> .\PrepareNode.ps1 -KubernetesVersion v1.17.0
+
+#>
+
+Param(
+    [parameter(Mandatory = $true, HelpMessage="Kubernetes version to use")]
+    [string] $KubernetesVersion
+)
+$ErrorActionPreference = 'Stop'
+
+function DownloadFile($destination, $source) {
+    Write-Host("Downloading $source to $destination")
+    curl.exe --fail -Lo $destination $source
```

Comment:
hm, i don't think curl is available on Windows by default. i remember having to download it when the last time i've tested.

Reply:
It's available by default from Windows 1803 on, so on every version that we're targeting

Reply:
interesting, you are probably right.